hazelcast/src/main/java/com/hazelcast/util/HashUtil.java

```diff
@@ -170,23 +175,36 @@ public static int MurmurHash3_x86_32_direct(long address, int offset, int len, i
         return h1;
     }
 
-    public static long MurmurHash3_x64_64(final byte[] data, int offset, int len) {
-        return MurmurHash3_x64_64(data, offset, len, DEFAULT_MURMUR_SEED);
+    public static int MurmurHash3_x86_32_direct(long base, int offset, int len) {
+        return MurmurHash3_x86_32_direct(base, offset, len, DEFAULT_MURMUR_SEED);
+    }
+
+    public static int MurmurHash3_x86_32_direct(long base, int offset, int len, int seed) {
+        return MurmurHash3_x86_32_direct(MEM, base, offset, len, seed);
+    }
+
+    public static int MurmurHash3_x86_32_direct(MemoryAccessor mem, long base, int offset, int len) {
+        return MurmurHash3_x86_32_direct(mem, base, offset, len, DEFAULT_MURMUR_SEED);
     }
 
+
     public static long MurmurHash3_x64_64(final byte[] data, int offset, int len, final int seed) {
```

Comment:
We have the same algo for 2 different cases
data[] and MemoryAccessor + we have a lot of abstractions how to link them
Can we may be have 1 Murmur calculator + ByteAccessStrategy ?